### PR TITLE
test(loader): refactor for component

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour

Refactor for <Loader> component to use new cypress-component-react framework for testing.

### Current behaviour

Currently we are using the regular cypress approach with BDD.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

- Run npx cypress open-ct to check if there is newly added test.file
 - Check if the loader.test.js file passed
 - Compare all tests were moved/refactored from .feature file -> .test.js file
 - Check if all unnecessary files were deleted, including .json file

### CODE REVIEW

File: src/components/loader/loader.test.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        "have.css",
+        "background-color",
+        color
+      );
+    });
+
+    it("should render Loader inside the Button component and be able to focus it", () => {
+      CypressMountWithProviders(<LoaderInsideButton />);
+
+      buttonDataComponent()
```
Comment (ZhuoyuJin) [inline src/components/loader/loader.test.js:100]: Shouldn't it be loaderInsideButton()?

Comment (DlgSHi) [reply src/components/loader/loader.test.js]: yes, you are right, fixed

**Review by ZhuoyuJin — DISMISSED** [review]

**Review by grabkowski — DISMISSED** [review]

File: src/components/loader/loader.test.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      [LOADER_SIZES[2], 128],
+    ])("should render Loader using %s as size in Button", (size, width) => {
+      CypressMountWithProviders(<LoaderInsideButton size={size} />);
+
+      loaderInsideButton()
+        .should("have.css", "height", "40px")
+        .and("have.css", "width", `${width}px`);
+    });
+
+    it("should render Loader inside the Button component", () => {
```
Comment (sianford) [inline src/components/loader/loader.test.js:57]: What is the purpose of this test case when there are already three preceding this one to test the Loader inside a button? The only difference I can see is that the background colour is checked. Could this assertion be added to the previous Loader in button test cases and remove this one?

Comment (DlgSHi) [reply src/components/loader/loader.test.js:57]: as per testing theory we suppose to check one assertion in one TS (at least the same parameters) It the test above we are checking the `size` but here only the colour :)

Comment (sianford) [reply src/components/loader/loader.test.js:57]: Ok, could you amend the test to specify "should render Loader inside the Button component with correct background colour" then please so it's clear what its purpose is as when you see it running it looks like it's a duplicate test case from the test name.

Comment (DlgSHi) [reply src/components/loader/loader.test.js]: fixed

**Review by sianford — APPROVED** [review]

**Review by grabkowski — APPROVED** [review]

**Pull request merged**